monkey/infection_monkey/exploit/wmiexec.py

```diff
@@ -32,6 +32,9 @@ def _exploit_host(self) -> ExploiterResultData:
         creds = generate_brute_force_combinations(self.options["credentials"])
 
         for user, password, lm_hash, ntlm_hash in creds:
+            if self.is_interrupted():
+                return self.exploit_result
```

Comment:
Use `infection_monkey.utils.threading.interruptable_iter()`

```suggestion
        for user, password, lm_hash, ntlm_hash in interruptable_iter(creds, self._interrupt):
```